tx-pool/src/pool.rs

```diff
@@ -465,6 +529,114 @@ impl TxPool {
         (entries, size, cycles)
     }
 
+    pub(crate) fn check_rbf(
+        &self,
+        snapshot: &Snapshot,
+        rtx: &ResolvedTransaction,
+        conflicts: &HashSet<ProposalShortId>,
+        fee: Capacity,
+        tx_size: usize,
+    ) -> Result<(), Reject> {
+        assert!(self.enable_rbf());
+        assert!(!conflicts.is_empty());
+
+        let short_id = rtx.transaction.proposal_short_id();
+        // Rule #4, new tx' fee need to higher than min_rbf_fee computed from the tx_pool configuration
+        // Rule #3, new tx's fee need to higher than conflicts, here we only check the root tx
+        if let Some(min_replace_fee) = self.calculate_min_replace_fee(conflicts, tx_size) {
+            if fee < min_replace_fee {
+                return Err(Reject::RBFRejected(format!(
+                    "Tx's current fee is {}, expect it to >= {} to replace old txs",
+                    fee, min_replace_fee,
+                )));
+            }
+        } else {
+            panic!("calculate_min_replace_fee must success");
```

Comment:
How about return an `err` to upper level, don't panic.

Reply:
Sounds reasonable.